src/rootPages/Designer/properties/process/ABProcessTaskServiceInsertRecord.js

```diff
@@ -0,0 +1,470 @@
+            value: L("[Primary Key]"),
+         });
+
+         object.fields().forEach((f) => {
+            // Populate fields of linked data source
+            if (f.isConnection) {
+               const linkDS = f.datasourceLink;
+               if (linkDS) {
+                  result.push({
+                     id: `${f.id}|PK`,
+                     value: `${f.label} -> ${L("[Primary Key]")}`,
+                  });
+
+                  linkDS.fields().forEach((linkF) => {
+                     result.push({
+                        id: `${f.id}|${linkF.id}`,
+                        value: `${f.label} -> ${linkF.label}`,
+                     });
+                  });
+               }
+            } else {
+               result.push({
+                  id: f.id,
+                  value: f.label,
+               });
+            }
+         });
+
+         return result;
+      }
+
+      refreshFieldValues(objectID) {
+         const ids = this.ids;
+
+         const $fieldValues = $$(ids.fieldValues);
+         if (!$fieldValues) return;
+
+         // clear form
+         webix.ui([], $fieldValues);
+
+         const object = this.AB.objectByID(objectID ?? this.element.objectID);
+         if (!object) return;
+
+         // Pull object & fields of start step
+         const startElemObj = this.element.objectOfStartElement;
+         const startElemObjFields = startElemObj
+            ? this.getFieldOptions(startElemObj)
+            : [];
+
+         // Pull object & fields of previous step
+         const prevElemObj = this.objectOfPrevElement;
+         let prevElemObjFields = [];
+         if (prevElemObj) {
+            prevElemObjFields = this.getFieldOptions(prevElemObj);
+         }
+
+         const setOptions = [
+            { id: 0, value: L("Not Set") },
+            { id: 1, value: L("Set by custom value") },
+            {
+               id: 2,
+               value: L("Set by the root data [{0}]", [
+                  startElemObj ? startElemObj.label : "",
+               ]),
+            },
+            {
+               id: 3,
+               value: L("Set by previous step data [{0}]", [
+                  prevElemObj ? prevElemObj.label : "",
+               ]),
+            },
+            {
+               id: 4,
+               value: L("Set by formula format"),
+            },
+         ];
+
+         let repeatObjectFields = [];
+         const fieldRepeat = this.element.fieldRepeat;
+
+         if (fieldRepeat && fieldRepeat.datasourceLink) {
+            setOptions.push({
+               id: 5,
+               value: L("Set by the instance [{0}]", [
+                  this.fieldRepeat ? this.element.fieldRepeat.label : "",
+               ]),
+            });
+
+            repeatObjectFields = this.getFieldOptions(
+               fieldRepeat.datasourceLink
+            );
+         }
+
+         setOptions.push({
+            id: 6,
+            value: `Set by the paremeter of a Query task`,
```

Comment:
this should be a `L("Set by the parameter of a Query task")`   <--- also notice spelling of `parameter`